src/shogun/machine/Machine.h

```diff
@@ -328,11 +341,38 @@ class CMachine : public CStoppableSGObject
 		 */
 		virtual bool train_machine(CFeatures* data=NULL)
 		{
-			SG_ERROR("train_machine is not yet implemented for %s!\n",
-					get_name());
+			bool result=false;
+			
+			switch (data->get_feature_class())
+			{
+				case C_DENSE:
+					result=train_with_dense(data);
+					break;
+				case C_STRING:
+					result=train_with_string(data);
+					break;
+				default:
+					SG_ERROR("train_machine is not yet implemented for %s!\n",
```

Comment:
the error message should change and say that.
"Training %s with %s is not supported.\n", get_name(), data->get_name()